### ISSUE TITLE
Blank Publish Page

### ISSUE THREAD

**Issue by ErLaby** [formtools/core#692]
Hi,
Looking for some breadcrumbs on what to look for.. I am new to Form Tools and this forum.

I am working to create my first form. When I go to the publish tab in the form creation, it comes up with a blank page.  I have to believe I've done something basic, wrong.
The page link for publish tab is: localhost/formtools/admin/forms/edit/?form_id5

I am using the default firefox browser that ships with ubuntu.

**Comment by kubajal** [comment]
Hi,
that sound very odd, Are you using Form Builder? I recommend to strictly follow the steps in this [Publishing your First Form](https://docs.formtools.org/modules/form_builder/usage/tutorials/publishing_first_form/) tutorial. Creating a minimal form should be extremely simple.

**Comment by ErLaby** [comment]
Yes, I followed it exactly to use FormBuilder and when I get to the Publish, the screen is blank. I am using v 3.0.20 with Form Builder 2.0.10. This is on Ubuntu.

Every time I try to build a form, no matter how simple, the "Publish" button always returns a blank page.

**Comment by kubajal** [comment]
Do you see any errors? Can you please open Developer Tools  (https://developer.mozilla.org/en-US/docs/tools/Tools_Toolbox) on Mozilla, go to Console and post here any errors that you see (preferably taking screenshots)?

**Comment by ErLaby** [comment]
Hi, Here's a screen shot. As you can see, Console is blank. The "inspector" just shows the following:

<html>
   <head></head>
  <body></body>
</html>
![2020-04-13_16-58-02](https://user-images.githubusercontent.com/63606019/79160640-1c3da080-7da8-11ea-8a53-85b24f809030.jpg)

**Comment by kubajal** [comment]
That looks very strange. Are you sure you downloaded and unpacked all files? In particular does the file _{FormTools root}/admin/forms/edit/index.php_ even exist? Also, can you post screenshots of every step you make that lead to this blank page?

**Comment by ErLaby** [comment]
Hi,
Yes, the index.php file is located in the [root]/admin/forms/edit/ directory. For the form set up, I followed the tutorial example exactly.

**Comment by ErLaby** [comment]
The file I downloaded was the default zip file from formtools.org. I unpacked it all to /var/www/html/formtools

I've tried to create very basic forms using all 3 types and have made several attempts to use the Form Builder option:

Form Name: Test
Number of fields: 2
Access: Administrator Only
<<Add form>>

Main:
"Your Form Builder form has been created! Use these tabs to configure and publish your form on your website.

Status: Online
Form Name: test
Form Type: Form Builder
Access: Administrator Only
Delete associated files when deleting submissions: Yes
Strip tags in form submissions: Yes
Edit Submission Label: Edit Submission
Add Submission Button: Add >>

Then I press <<Update>>
Then I go to the "Publish" tab and it is blank every time

**Comment by ErLaby** [comment]
I just tried using a different browser, and I get: localhost is currently unable to handle this request HTTP ERROR 500

**Comment by kubajal** [comment]
Good move. Can you find something similar to the Toolbox/Console in Mozilla to check what exactly causes the HTTP 500 error that you get and post it here? Also, can you make sure that the file _[root]/modules/form_builder/admin/tab_publish.php_ exists? I suspect that your Form Builder is corrupt.

**Comment by kubajal** [comment]
Can you also execute _ls -all_ in /var/www/formtools/themes/default/cache?

**Comment by ErLaby** [comment]
Hi,
tab_publish.php exists

/var/www/formtools/themes/default/cache <=Does not exist

**Comment by kubajal** [comment]
Ok, please execute the following and see if it helps:
```
cd /var/www/formtools/themes/default
mkdir cache
chmod 750 cache
```

**Comment by ErLaby** [comment]
I ran the commands exactly. The cache directory is now present. However, it still did not help. I am met with the exact same results, when I edit a form (any form) I still get a blank "Publish" page.

**Comment by kubajal** [comment]
OK, who is the owner of the directory? can you `ls -all` in eg /var/www/formtools/themes/default and paste here the output?

**Comment by ErLaby** [comment]
total 96
drwxr-xr-x 9 root root 4096 Apr 13 22:25 .
drwxrwxrwx 5 root root 4096 Nov  9 22:22 ..
drwxr-xr-x 2 root root 4096 Nov  9 22:21 about
drwxr-xr-x 8 root root 4096 Nov  9 22:21 admin
drwxr-x--- 2 root root 4096 Apr 13 22:25 cache
drwxr-xr-x 4 root root 4096 Nov  9 22:21 clients
drwxr-xr-x 3 root root 4096 Nov  9 22:21 css
-rw-r--r-- 1 root root 2639 Nov  9 22:21 error.tpl
-rw-r--r-- 1 root root  292 Nov  9 22:21 footer.tpl
-rw-r--r-- 1 root root 1033 Nov  9 22:21 forget_password.tpl
-rw-r--r-- 1 root root 3457 Nov  9 22:21 header.tpl
drwxr-xr-x 2 root root 4096 Nov  9 22:21 images
-rw-r--r-- 1 root root 1669 Nov  9 22:21 index.tpl
-rw-r--r-- 1 root root 1721 Nov  9 22:21 js_pagination.tpl
-rw-r--r-- 1 root root  982 Nov  9 22:21 menu.tpl
-rw-r--r-- 1 root root 2247 Nov  9 22:21 messages.tpl
-rw-r--r-- 1 root root  463 Nov  9 22:21 module_menu.tpl
-rw-r--r-- 1 root root  278 Nov  9 22:21 modules_footer.tpl
-rw-r--r-- 1 root root 4243 Nov  9 22:21 modules_header.tpl
-rw-r--r-- 1 root root 1917 Nov  9 22:21 pagination.tpl
drwxr-xr-x 2 root root 4096 Nov  9 22:21 scripts
-rw-r--r-- 1 root root    8 Nov  9 22:21 tabset_close.tpl
-rw-r--r-- 1 root root 1576 Nov  9 22:21 tabset_open.tpl

**Comment by kubajal** [comment]
Yea, it was helpful, there is a problem with permissions. I just checked what permissions I have in my Formtools installation and they look like this:
![image](https://user-images.githubusercontent.com/9532517/79169433-bd4e4a80-7dec-11ea-823e-8e05f0e00623.png)
![image](https://user-images.githubusercontent.com/9532517/79169444-c7704900-7dec-11ea-8df0-c9476a628020.png)


I guess we should aim for the same for you. Can you execute:
```
cd /var/www/formtools/themes/default
mkdir cache <---- (if it not exists yet)
chmod 770 cache
```
Note there is 770 now.

**Comment by ErLaby** [comment]
It is working! Thank you this is a huge help! Cheers! Someday I wish to be able to be so kind as you as to be able to solve problems on forums like these!

**Comment by kubajal** [comment]
No worries, glad it helped.

**Comment by apmuthu** [comment]
For Ubuntu / Debian:

````
cd /var/www/webroot
wget -c --no-check-certificate \
    "https://formtools.org/download_file.php?file_type=core&file=FormTools3.0.20-20191109.zip" \
	-O FormTools3.0.20-20191109.zip

unzip FormTools3.0.20-20191109.zip
cd formtools
chown -R www-data:www-data upload cache
# only the folder and not the files and folders within it:
chown www-data:www-data global
````
